node/narwhal/src/primary.rs

```diff
@@ -1053,3 +1053,317 @@ impl<N: Network> Primary<N> {
+            (accounts, Committee::<CurrentNetwork>::new(1, members).unwrap())
+        };
+
+        let account = accounts.first().unwrap().1.clone();
+        let storage = Storage::new(committee, 10);
+        let ledger = Arc::new(MockLedgerService::new());
+
+        // Initialize the primary.
+        let mut primary = Primary::new(account, storage, ledger, None, None).unwrap();
+
+        // Construct a worker instance.
+        primary.workers = Arc::from([Worker::new(
+            0, // id
+            Arc::new(primary.gateway.clone()),
+            primary.storage.clone(),
+            primary.ledger.clone(),
+            primary.proposed_batch.clone(),
+        )
+        .unwrap()]);
+
+        (primary, accounts)
+    }
+
+    // Creates a mock solution.
+    fn sample_unconfirmed_solution(
+        rng: &mut TestRng,
+    ) -> (PuzzleCommitment<CurrentNetwork>, Data<ProverSolution<CurrentNetwork>>) {
+        // Sample a random fake puzzle commitment.
+        let affine = rng.gen();
+        let commitment = PuzzleCommitment::<CurrentNetwork>::from_g1_affine(affine);
+        // Sample random fake solution bytes.
+        let mut vec = vec![0u8; 1024];
+        rng.fill_bytes(&mut vec);
+        let solution = Data::Buffer(Bytes::from(vec));
+        // Return the ID and solution.
+        (commitment, solution)
+    }
+
+    // Creates a mock transaction.
+    fn sample_unconfirmed_transaction(
+        rng: &mut TestRng,
+    ) -> (<CurrentNetwork as Network>::TransactionID, Data<Transaction<CurrentNetwork>>) {
+        // Sample a random fake transaction ID.
+        let id = Field::<CurrentNetwork>::rand(rng).into();
+        // Sample random fake transaction bytes.
+        let mut vec = vec![0u8; 1024];
+        rng.fill_bytes(&mut vec);
+        let transaction = Data::Buffer(Bytes::from(vec));
+        // Return the ID and transaction.
+        (id, transaction)
+    }
+
+    // Creates a batch proposal with one solution and one transaction.
+    fn create_test_proposal(
+        author: &Account<CurrentNetwork>,
+        current_committee: Committee<CurrentNetwork>,
+        round: u64,
+        timestamp: i64,
+        rng: &mut TestRng,
+    ) -> Proposal<CurrentNetwork> {
+        let (solution_commitment, solution) = sample_unconfirmed_solution(rng);
+        let (transaction_id, transaction) = sample_unconfirmed_transaction(rng);
+
+        // Retrieve the private key.
+        let private_key = author.private_key();
+        // Prepare the transmission IDs.
+        let transmission_ids = [solution_commitment.into(), (&transaction_id).into()].into();
+        let transmissions = [
+            (solution_commitment.into(), Transmission::Solution(solution)),
+            ((&transaction_id).into(), Transmission::Transaction(transaction)),
+        ]
+        .into();
+        // Prepare the certificate IDs.
+        let certificate_ids = Default::default();
+        // Sign the batch header.
+        let batch_header =
```

Comment:
There is also a `snarkos_node_narwhal:test_helpers::sample_batch_header(rng);`. Can that be used?

Reply:
I can find `sample_batch_certificate` but not `sample_batch_header`?

Reply:
Also can't use it unfortunately, I need the transmissions to create a proposal, not just the transmission IDs, good shout though 👍